server/src/testFixtures/java/io/crate/testing/TestingHelpers.java

```diff
@@ -78,6 +81,37 @@
 
 public class TestingHelpers {
 
+    private static final boolean HAS_RANDOMIZED_CONTEXT;
+
+    static {
+        boolean maybeHasRandomizedContext;
+        try {
+            Class.forName("com.carrotsearch.randomizedtesting.RandomizedContext");
+            maybeHasRandomizedContext = true;
+        } catch (Exception e) {
+            maybeHasRandomizedContext = false;
+        }
+        HAS_RANDOMIZED_CONTEXT = maybeHasRandomizedContext;
+    }
+
+    /**
+     * Creates an empty, temporary folder using the {@link com.carrotsearch.randomizedtesting.RandomizedContext}
+     * helper if the class is loaded.
+     * Use this method instead of {@link CrateLuceneTestCase#createTempDir()} in components which may not run under
+     * the {@link com.carrotsearch.randomizedtesting.RandomizedRunner} like e.g. JMH benchmarks.
+     *
+     */
+    public static Path createTempDir() {
+        if (HAS_RANDOMIZED_CONTEXT) {
+            return CrateLuceneTestCase.createTempDir();
+        }
+        try {
+            return Files.createTempDirectory(null);
```

Comment:
Afaik the test case variant makes sure that the temporary folder is cleaned up (at least in the success case).
This would leave the folders hanging around. Couldn't we initialize the random context for the benchmarks too?

Reply:
Right, I could not found a way to initialize random context with JMH as it will raise exceptions when not run under the `RandomizedRunner`. Any idea?

Reply:
Maybe removing the folders in a jvm shutdown hook would be good enough for the jmh case?

Otherwise we'd have to re-create something like the randomizedRunner+context for benchmarks or just accept the folder leaks.